### DISCUSSION TITLE
AOV interface similar to Eevee?

### DISCUSSION THREAD

**Discussion by rosstheblenderanimator** [bnpr/Malt#594] [Feedback & Feature Requests]
I was able to render out this car tunnel https://bsky.app/profile/rossbergin.bsky.social/post/3lcfpt24f4c2l animation into different layers by doing this(below). And it does not show up in the viewport so if I wanted to adjust the line art or bloom I would have to undo that. I could not figure out how to do a mist pass as I wanted to do depth of field. How hard would it be to add similar interface Eevee instead of having to do the manual set up? Would it better for it be a plugin for Malt? I was thinking of trying to do it myself at some point.
![car tunnel animation final 1](https://github.com/user-attachments/assets/a12c29d1-c2dd-4b7c-9414-b147e97a45bb)
![car tunnel animation final 2](https://github.com/user-attachments/assets/627a0c74-1ccc-4086-953a-13aafa8b9b27)
![car tunnel animation final 3](https://github.com/user-attachments/assets/1cf579d6-2b33-4e65-91f6-ef7de43b3bbc)
![car tunnel animation final 4](https://github.com/user-attachments/assets/d0fe2899-2b65-42d7-b62f-0e59a5038b3d)
Noctis whole showed me how to this.
![screen](https://github.com/user-attachments/assets/3d492976-cbff-4a52-af1f-c6607953de73)
This is what I want

**Comment by pragma37** [comment]
This is the price to pay for the extra flexibility that Malt offers.
For example, you can have more than one MainPass node. With an EEEVEE-like setup, how should it be decided which set of AOVs are exported?

I understand that if your goal is sending AOVs to the Blender compositor, declaring them in 3 different places may be annoying, but I can't think of a way of simplifying this that would work well with the way the default Malt's pipeline works.

On the other hand, the EEVEE/Cycles workflow requires manually adding an AOV node and writing its name to every single Material that uses it, so it also has its downsides.

**Reply by rosstheblenderanimator** [reply]
-With an EEEVEE-like setup, how should it be decided which set of AOVs are exported?
It depends on the project. This was my first time doing any proper compositing. I wanted to do a depth of field effect and have the truck in focus and car and background blurred.
![Screenshot 2024-12-08 164458](https://github.com/user-attachments/assets/53e9c50c-0972-4455-8904-6b6b927ce340)


-On the other hand, the EEVEE/Cycles workflow requires manually adding an AOV node and writing its name to every single Material that uses it, so it also has its downsides. 
You can use node groups and scripts.
I think more people would use Malt if it had AOVs support.
